ctl/waypoint/waypoint.go

```diff
@@ -0,0 +1,618 @@
+/*
+ * Copyright The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// NOTE: THE CODE IN THIS FILE IS MAINLY REFERENCED FROM ISTIOCTL WAYPOINT
+// SUBCOMMAND(https://github.com/istio/istio/tree/master/istioctl/pkg/waypoint)
+// AND ADAPTED FOR KMESH.
+
+package waypoint
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"sync"
+	"text/tabwriter"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"istio.io/api/label"
+	"istio.io/istio/pilot/pkg/model/kstatus"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/config/schema/gvk"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/util/sets"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gateway "sigs.k8s.io/gateway-api/apis/v1"
+
+	"kmesh.net/kmesh/ctl/utils"
+)
+
+var (
+	revision      = ""
+	waitReady     bool
+	allNamespaces bool
+	namespace     = ""
+
+	deleteAll bool
+
+	trafficType       = ""
+	validTrafficTypes = sets.New(constants.ServiceTraffic, constants.WorkloadTraffic, constants.AllTraffic, constants.NoTraffic)
+
+	DataplaneModeKmesh               = "Kmesh"
+	KmeshUseWaypointLabel            = "istio.io/use-waypoint"
+	KmeshWaypointForTrafficTypeLabel = "istio.io/waypoint-for"
+
+	WaypointImageAnnotation = "sidecar.istio.io/proxyImage"
+	KmeshWaypointImage      = "ghcr.io/kmesh-net/waypoint:latest"
```

Comment:
nit：If it in 0.5.0 release, is waypoint version also fixed to 0.5.0?

Reply:
Good catch!

It should be the same as the release version, but it should also be configurable, because we prefer to use the latest image during dev.